Fix ParserInterface implementations

equinoxmatt opened the pull request:
All the parsers (except ini) were not compliant with the ParserInterface.  Most of them returned array|null, rather than array defined in the interface.  Rather than fixing the parsers, (array) casts were being used in the consuming code (Config).

This PR:
- Makes all parsers compliant with the ParserInterface.
- Returns empty array in cases where null is returned from the parser parse() function.
- Removes (array) casting.
- General cleanup (remove unused params, docblocks etc).

equinoxmatt committed:
Make Yaml parser compliant with ParserInterface

Also do some clean up (unused params, docblocks, typos)

equinoxmatt committed:
Make XML parser compliant with ParserInterface

equinoxmatt committed:
Make PHP parser ParserInterface compliant

equinoxmatt committed:
Make json parser compliant with Parser Interface

equinoxmatt committed:
Remove array casts from parseString and parseFile calls

equinoxmatt committed:
Fix incorrect docblock

filips123 reviewed (changes requested):
It would probably be better to use array casting instead of ternary operators in all parsers. It would be cleaner and only one line instead of two.

So code will be:

```php
return (array) $this->parse($data, $filename);
```

equinoxmatt committed:
Cast to array

equinoxmatt commented:
> It would probably be better to use array casting instead of ternary operators in all parsers. It would be cleaner and only one line instead of two.
> 
> So code will be:
> 
> ```
> return (array) $this->parse($data, $filename);
> ```

@filips123 I have addressed your comments now 👍

filips123 commented:
Thanks!
